### PR TITLE
Created a method to send information from the reservation service to the email service

### PR DESCRIPTION
**Pull request by aldorotheo**
Created an object to hold a JSON representation of what's needed for the email. Created a function to retrieve users by id in the UserRepository and UserService. Created a method to post to the email service in the ReservationService. Added the method call to the ReservationController to call the post method when saving a reservation.
Austin D.

### CODE REVIEW

**Comment by aldorotheo** [comment]
Method successfully executed and email was successfully sent from the resource.service.dummy email to the slack user's email upon creating a reservation.

File: src/main/java/com/revature/controllers/ReservationController.java [hunk 1]
```diff
@@ -182,6 +190,8 @@ public Reservation saveReservation(@RequestBody Reservation reservation) {
 	 */
 	@PostMapping("cancel")
 	public int cancelReservation(@RequestParam int id) {
+		Reservation reservation = reservationService.getReservationById(id);
```
[the remark this answers is not in the document]
Comment (aldorotheo) [reply src/main/java/com/revature/controllers/ReservationController.java]: Currently this assignment is not being used. Can just remove it until it has a reason to be in this method.

File: src/main/java/com/revature/controllers/ReservationController.java [hunk 2]
```diff
@@ -67,12 +73,13 @@ public ReservationController(ReservationService reservationService) {
...
 	 */
 	private List<Resource> getResourcesByCampus(int campusId) {
 
 		RestTemplate restTemplate = new RestTemplate();
 		ResponseEntity<List<Resource>> response = restTemplate.exchange(
-				this.uri + "/campus/" + String.valueOf(campusId), HttpMethod.GET, null,
+				URI.create(this.uri + "/campus/" + String.valueOf(campusId)), HttpMethod.GET, null,
 				new ParameterizedTypeReference<List<Resource>>() {
 				});
 		List<Resource> resources = response.getBody();
```
[the remark this answers is not in the document]
Comment (aldorotheo) [reply src/main/java/com/revature/controllers/ReservationController.java]: With this code, we could immediately return response.getBody(). There's no real reason to use the resources variable.

File: src/main/java/com/revature/controllers/ReservationController.java [hunk 3]
```diff
@@ -67,12 +73,13 @@ public ReservationController(ReservationService reservationService) {
...
 	 * @param campusId The identification number for a campus.
 	 * @return A list of resources. 
+	 * @author Jaron 1811-Java-Nick 1/2/19
 	 */
 	private List<Resource> getResourcesByCampus(int campusId) {
 
 		RestTemplate restTemplate = new RestTemplate();
 		ResponseEntity<List<Resource>> response = restTemplate.exchange(
-				this.uri + "/campus/" + String.valueOf(campusId), HttpMethod.GET, null,
+				URI.create(this.uri + "/campus/" + String.valueOf(campusId)), HttpMethod.GET, null,
```
[the remark this answers is not in the document]
Comment (aldorotheo) [reply src/main/java/com/revature/controllers/ReservationController.java]: Don't even need the String.valueOf(), can just do + campusId

File: src/main/java/com/revature/controllers/ReservationController.java [hunk 4]
```diff
@@ -20,8 +21,10 @@
 import com.revature.enumerations.Purpose;
 import com.revature.enumerations.Type;
 import com.revature.models.Reservation;
+import com.revature.models.ReservationEmail;
```
[the remark this answers is not in the document]
Comment (aldorotheo) [reply src/main/java/com/revature/controllers/ReservationController.java]: Remove any imports that aren't used if we change the code to remove why they were there.

**Review by nuklearpineapple — APPROVED** [review]
Code smells

**Comment by aldorotheo** [comment]
Also added the method and method call to send an email confirming that a reservation for properly cancelled when the cancellation is called.

**Pull request merged**